### PR TITLE
fix: Are auto-teleported funds preferred over funds on ahp?

### LINKED ISSUE #9962: Are auto-teleported funds preferred over funds on ahp?
One of the people who lost funds in auto teleport was [this guy](https://github.com/kodadot/nft-gallery/issues/9943#issuecomment-2032455422). Sadly, he lost his funds twice. After losing his funds the first time, there was  1.1DOT added to his account, **but this time on AHP directly**. So he had something newly added on Polkadot relay chain, but he also had 1.1 DOT on ahp. He wanted to mint 1 NFT (Drops: [Shapes](https://kodadot.xyz/ahp/drops/shapes)). He was asked to use auto teleport, and that's when he lost his funds from the relay chain a second time. 

Extrinsic: https://polkadot.subscan.io/extrinsic/20164870-3 
Address: 14dHRg2P8NQfg68MktZjYeWbgvXEUH4KsvXEoxByeryMXPWB
Amount: 1.6
Twitter: https://twitter.com/rickybobby440

My question is, existential minimum on ahp is 0.1, how is it possible that autoteleport was activated even though he had enough funds in his wallet on ahp? (1.1DOT)

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

see https://github.com/kodadot/nft-gallery/issues/9962#issuecomment-2034106409

- Teleport ED will be applied when performing XCM calls , for both the teleport page and auto-teleport 

@prury now if you have 1 DOT and you try to mint/buy a NFT that costs 0.5 DOT  you'll be able to, and the auto-teleport won't be triggered , since it will use the normal ED of 0.01 DOT

 ❗️❗️❗️ **it should ONLY** use the teleport whenever a XCM call it's being performed , pls check this thoroughly  

## Separate EDs

teleport ED:
ahk\rmrk\ksm: 0.00066 KSM
ahp:  0.5 DOT
dot: 1.5 DOT

normal ED:
ahk\rmrk\ksm: 0.00033 KSM 
ahp:  0.1 DOT
dot: 1 DOT

- [x] Closes #9962

## Needs QA check

- @kodadot/qa-guild please review

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)

## Screenshot 📸

- [ ] My fix has changed **something** on UI;

### CODE REVIEW

**Comment by yangwao** [comment]
@hassnian is this something we should have in production today?

File: libs/static/src/chains.ts [hunk 1]
```diff
@@ -93,13 +93,23 @@ export const chainList = (): Option[] => {
...
+  imx: 0,
+  base: 0,
+}
+
+export const existentialDeposit: Record<Prefix, number> = {
+  ksm: 333333333,
+  rmrk: 333333333,
+  ahk: 333333333,
+  dot: 10000000000,
+  ahp: 1000000000,
```
Comment (vikiival) [inline libs/static/src/chains.ts:112]: Screenshot 2024-04-03 at 11 42 48

**Comment by hassnian** [comment]
> @hassnian is this something we should have in production today?

With this PR no unnecessary auto teleport will be triggered , since now we have 2 sets of ED


If you have O.8 DOT on ahp a try to mint/buy (let's say that the price is 0.5 DOT) since the ED is 0.1 DOT , you'll be able to buy/mint

I think I would be great if we could have it by today since we have a drop

Not a must but great 

Tested ofc

File: libs/static/src/chains.ts (hunk 1 above)
Comment (hassnian) [reply libs/static/src/chains.ts:112]: hmm it's  0.1

CleanShot 2024-04-04 at 16 42 42@2x

**Comment by prury** [comment]
next drop it is 😄

File: libs/static/src/chains.ts (hunk 1 above)
Comment (prury) [reply libs/static/src/chains.ts:112]: now you two got me thinking, is it the true polkadot assethub existential deposit 0.1 or 0.01? 
i can find both infos on polkadot docs

**Comment by yangwao** [comment]
cc @prury so is this tested?

File: libs/static/src/chains.ts (hunk 1 above)
Comment (vikiival) [reply libs/static/src/chains.ts:112]: [This one is the current deposit](https://github.com/kodadot/nft-gallery/pull/9972#discussion_r1551483528)

Comment (hassnian) [reply libs/static/src/chains.ts]: updated 4ab191f

**Comment by vikiival** [comment]
I would keep the ED as it was bfr [4ab191f](https://github.com/kodadot/nft-gallery/pull/9972/commits/4ab191f4b30edbbe76716ef6905776b20f6fbf75)

**Comment by hassnian** [comment]
> I would keep the ED as it was bfr [4ab191f](https://github.com/kodadot/nft-gallery/pull/9972/commits/4ab191f4b30edbbe76716ef6905776b20f6fbf75)

🫡  3353b00

**Review by daiagi — APPROVED** [review]

**Review by vikiival — APPROVED** [review]

**Comment by prury** [comment]
pay 35 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $35 @ 8.26 USD/DOT ~ 4.237 $DOT
🧗 [16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD](https://polkadot.subscan.io/account/16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)
🔗 [0x4035c7d20afff953301820da85dc752d6baa8cba852635909e5e8b06c34f0cb8](https://polkadot.subscan.io/extrinsic/0x4035c7d20afff953301820da85dc752d6baa8cba852635909e5e8b06c34f0cb8)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**